pkg/mobile/metrics/fh-sync-server.go

```diff
@@ -0,0 +1,347 @@
+				From            time.Time `json:"from"`
+				End             time.Time `json:"end"`
+			} `json:"doReadDatasetClient"`
+			DoUpdateDatasetClientWithRecords struct {
+				Current         string    `json:"current"`
+				Max             string    `json:"max"`
+				Min             string    `json:"min"`
+				Average         string    `json:"average"`
+				NumberOfRecords int64     `json:"numberOfRecords"`
+				From            time.Time `json:"from"`
+				End             time.Time `json:"end"`
+			} `json:"doUpdateDatasetClientWithRecords"`
+			DoReadDatasetClientWithRecordsUseCache struct {
+				Current         string    `json:"current"`
+				Max             string    `json:"max"`
+				Min             string    `json:"min"`
+				Average         string    `json:"average"`
+				NumberOfRecords int64     `json:"numberOfRecords"`
+				From            time.Time `json:"from"`
+				End             time.Time `json:"end"`
+			} `json:"doReadDatasetClientWithRecordsUseCache"`
+			DoSaveUpdate struct {
+				Current         string    `json:"current"`
+				Max             string    `json:"max"`
+				Min             string    `json:"min"`
+				Average         string    `json:"average"`
+				NumberOfRecords int64     `json:"numberOfRecords"`
+				From            time.Time `json:"from"`
+				End             time.Time `json:"end"`
+			} `json:"doSaveUpdate"`
+		} `json:"Mongodb Operation Time"`
+	} `json:"metrics"`
+}
+
+type FhSyncServer struct {
+	requestBuilder     mobile.HTTPRequesterBuilder
+	tokenClientBuilder mobile.TokenScopedClientBuilder
+	ServiceName        string
+	logger             *logrus.Logger
+}
+
+func NewFhSyncServer(rbuilder mobile.HTTPRequesterBuilder, tokenCBuilder mobile.TokenScopedClientBuilder, l *logrus.Logger) *FhSyncServer {
+	return &FhSyncServer{requestBuilder: rbuilder, tokenClientBuilder: tokenCBuilder, ServiceName: "fh-sync-server", logger: l}
+}
+
+// Gather will retrieve varous metrics from fh-sync-server
+func (ss *FhSyncServer) Gather() ([]*metric, error) {
+	svc, err := ss.tokenClientBuilder.UseDefaultSAToken().MobileServiceCruder("")
+	if err != nil {
+		return nil, errors.Wrap(err, "fh-sync-server gather failed to create svcruder")
+	}
+	ssServices, err := svc.List(func(attrs mobile.Attributer) bool {
+		return attrs.GetName() == ss.ServiceName
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "fh-sync-server gather failed to list existing services")
+	}
+	if len(ssServices) == 0 {
+		return nil, errors.New(" no fh-sync-server service present")
+	}
+	ssService := ssServices[0] //TODO deal with more than one
+	//TODO get protocol from secret
+	host := ssService.Host
+	now := time.Now()
+
+	stats, err := ss.getStats(host)
+	if err != nil {
+		return nil, errors.Wrap(err, "fh-sync-server gather failed to list existing services")
+	}
+
+	var ssMetrics = []*metric{}
+	if nil != stats {
+		// API Process Times
+		syncAPIProcessTime, err := stringToInt64(stats.Metrics.APIProcessTime.Sync.Current)
+		if err == nil {
```

Comment:
are we just ignoring errors here? Maybe what we could do is push any errors into a slice and after processing the stats, print any errors that occurred? Not a fan of ignoring the errors completely

Reply:
From the api callers point of view, the metric can't be parsed, so best to leave it out.
Lets add logging on the server like you suggest for cases where the metric couldn't be parsed.